frontend/express/app.js

```diff
@@ -397,6 +398,18 @@ app.set('views', __dirname + '/views');
 app.set('view engine', 'html');
 app.set('view options', {layout: false});
 
+// minify html if running production
+if (plugins.getConfig("frontend").production) {
+    app.use(minifyHTML({
+        override: true,
+        exception_url: false,
+        htmlMinifier: {
+            removeComments: true,
+            collapseWhitespace: true
+        }
+    }));
+}
+
```

Comment:
Sorry, for taking so long to review.
But there is only one thing that concerns me, the config should be runtime enabled/disabled.
But in this case this check is run only once on server start. And if I go and disable production mode, then it won't take any effect.

Would it be rather possible to have the production flag check inside the middleware to use it or skip it?

Reply:
@ar2rsawseen Thank you for the feedback. I just want to clarify where I should move the check.

By `middleware`, do you mean this one?

```
app.use(function(req, res, next) {
    add_headers(req, res);
    next();
});
```

So, the modification could be something like:

```
function check_minify() {
  if (plugins.getConfig("frontend").production) {
    app.use(minifyHTML({
        override: true,
        exception_url: false,
        htmlMinifier: {
            removeComments: true,
            collapseWhitespace: true
        }
    }));
  }
}

app.use(function(req, res, next) {
    add_headers(req, res);
    check_minify();
    next();
});
```
Let me know if I am going about this the right way. Thanks.

Reply:
With this, for each request (`app.use(function(req, res, next) {`) you will register a new middleware (`check_minify()` -> `app.use(minifyHTML(...`).

What you want to do is register your middleware only once (`app.use(function(req, res, next) {`), and in the `check_minify()` function, manually call a minification process (hopefully minifyHtml provide a way to do it manually).

Let me know if what I say is not clear.

Reply:
Middlewares are just functions, so does not matter if minifyHTML provides it or not, something like this should work 😄 
```
    var minify = minifyHTML({
        override: true,
        exception_url: false,
        htmlMinifier: {
            removeComments: true,
            collapseWhitespace: true
        }
    });
    app.use(function(req, res, next) {
        if (plugins.getConfig("frontend").production) {
            minify(req, res, next);
        }
        else {
          next(); 
       }
    });
```

Reply:
Yeah you can use the middleware as is.
However you don't want the `else` and run the `next()` all the time.